### PR TITLE
Fix Docker Compose setup

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Various fixes for the Docker Compose setup. Most importantly, the config-worker is added. See the commit messages for details.

### CODE REVIEW

File: config/local/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2023 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
```
Comment (sschuberth) [inline config/local/build.gradle.kts:2]: 2024?

File: config/local/src/main/kotlin/LocalConfigFileProvider.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    override fun contains(context: Context, path: Path) = configDir.resolve(path.path).isFile
+
+    override fun listFiles(context: Context, path: Path): Set<Path> {
+        val dir = configDir.resolve(path.path)
+
+        if (!dir.isDirectory) {
+            throw ConfigException("The provided path '${path.path}' does not refer a directory.", null)
+        }
+
+        return dir.listFiles()?.filter { it.isFile }?.mapTo(mutableSetOf()) { Path(it.path) }.orEmpty()
```
Comment (sschuberth) [inline config/local/src/main/kotlin/LocalConfigFileProvider.kt:77]: Maybe instead:

    return dir.walk().maxDepth(1).mapNotNullTo(mutableSetOf()) { file -> Path(file.path).takeIf { file.isFile } }

Comment (mnonnenmacher) [reply config/local/src/main/kotlin/LocalConfigFileProvider.kt]: `dir.walk()` provides a sequence of files, so I have kept the filter and the mapping separate for readability.

**Comment by Etsija** [comment]
So, as I understand, this should make the docker compose development environment work straight out of the box now?  How did you test it?

**Comment by mnonnenmacher** [comment]
> So, as I understand, this should make the docker compose development environment work straight out of the box now? Did you test it by any other means than running the unit tests?

We don't have any tests for the Compose setup yet, so I tested it manually by creating some ORT runs and verifying the results.

> I guess I could/should test this against my Server UI branch now - but as your PR is coming from a fork, maybe it would be simplest that I test against the ORT Server main branch after this PR gets merged and report the results? In case there are still some problems, I can always create issues/PRs about them.

You can still [checkout the PR](https://stackoverflow.com/a/30584951) to rebase your changes locally for testing.

**Comment by Etsija** [comment]
Ah, forgot about that, thanks for the tip!

**Comment by sschuberth** [comment]
> You can still [checkout the PR](https://stackoverflow.com/a/30584951)

Actually, I believe the easiest way to do this is via [GitHub CLI](https://cli.github.com/) and then `gh pr checkout 110`.

**Comment by Etsija** [comment]
When creating an ORT run:
```
fix-compose-config-worker-1     | 2024-04-19 08:03:39.150 [pool-1-thread-4] level=ERROR o.e.a.o.w.config.ConfigComponent - Config worker job failed for run '1'.
fix-compose-config-worker-1     | org.eclipse.apoapsis.ortserver.config.ConfigException: Could not find ConfigFileProviderFactory with name 'local-config' on classpath.
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManager$Companion$create$fileProviderSupplier$1.invoke(ConfigManager.kt:284)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManager$Companion$create$fileProviderSupplier$1.invoke(ConfigManager.kt:112)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManager$configFileProvider$2.invoke(ConfigManager.kt:164)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManager$configFileProvider$2.invoke(ConfigManager.kt:164)
fix-compose-config-worker-1     |       at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManager.getConfigFileProvider(ConfigManager.kt:164)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManager.access$getConfigFileProvider(ConfigManager.kt:39)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManager$resolveContext$1.invoke-1-1SFZQ(ConfigManager.kt:170)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManager$resolveContext$1.invoke(ConfigManager.kt:170)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManagerKt.wrapExceptions(ConfigManager.kt:271)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManagerKt.access$wrapExceptions(ConfigManager.kt:1)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.config.ConfigManager.resolveContext-wmjYzWQ(ConfigManager.kt:170)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.workers.config.ConfigWorker.run(ConfigWorker.kt:75)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.workers.config.ConfigComponent$endpointHandler$1$responsePayload$result$1.invokeSuspend(ConfigComponent.kt:56)
fix-compose-config-worker-1     |       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
fix-compose-config-worker-1     |       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
fix-compose-config-worker-1     |       at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:277)
fix-compose-config-worker-1     |       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:95)
fix-compose-config-worker-1     |       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:69)
fix-compose-config-worker-1     |       at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
fix-compose-config-worker-1     |       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:48)
fix-compose-config-worker-1     |       at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.workers.config.ConfigComponent$endpointHandler$1.invoke(ConfigComponent.kt:56)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.workers.config.ConfigComponent$endpointHandler$1.invoke(ConfigComponent.kt:51)
fix-compose-config-worker-1     |       at org.eclipse.apoapsis.ortserver.transport.rabbitmq.RabbitMqMessageReceiverFactory$createConsumer$1.handleDelivery(RabbitMqMessageReceiverFactory.kt:125)
fix-compose-config-worker-1     |       at com.rabbitmq.client.impl.ConsumerDispatcher$5.run(ConsumerDispatcher.java:149)
fix-compose-config-worker-1     |       at com.rabbitmq.client.impl.ConsumerWorkService$WorkPoolRunnable.run(ConsumerWorkService.java:111)
fix-compose-config-worker-1     |       at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
fix-compose-config-worker-1     |       at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
fix-compose-config-worker-1     |       at java.base/java.lang.Thread.run(Thread.java:840)
fix-compose-config-worker-1     | 2024-04-19 08:03:39.156 [pool-1-thread-4] level=INFO  o.e.a.o.transport.MessagePublisher - Sending 'ConfigWorkerError' message to 'OrchestratorEndpoint'. TraceID: '0f5bfcf8-a8bb-44ae-a23b-c9cb3842085f'.
fix-compose-config-worker-1     | 2024-04-19 08:03:39.157 [pool-1-thread-4] level=INFO  o.e.a.o.t.MessageSenderFactory - Creating a MessageSender of type 'rabbitMQ' for endpoint 'orchestrator'.
fix-compose-config-worker-1     | 2024-04-19 08:03:39.162 [pool-1-thread-4] level=INFO  o.e.a.o.t.r.RabbitMqMessageSenderFactory - Creating RabbitMQ sender for endpoint 'orchestrator'.
fix-compose-config-worker-1     | 2024-04-19 08:03:39.163 [pool-1-thread-4] level=INFO  o.e.a.o.t.r.RabbitMqMessageSenderFactory - RabbitMQ server URI: 'amqp://rabbitmq:5672'
fix-compose-config-worker-1     | 2024-04-19 08:03:39.163 [pool-1-thread-4] level=INFO  o.e.a.o.t.r.RabbitMqMessageSenderFactory - RabbitMQ user: 'admin'
fix-compose-config-worker-1     | 2024-04-19 08:03:39.163 [pool-1-thread-4] level=INFO  o.e.a.o.t.r.RabbitMqMessageSenderFactory - RabbitMQ queue: 'orchestrator_queue'
fix-compose-orchestrator-1      | 2024-04-19 08:03:39.208 [pool-1-thread-5] level=INFO  o.e.a.o.orchestrator.EntrypointKt - Received 'ConfigWorkerError' message. TraceID: '0f5bfcf8-a8bb-44ae-a23b-c9cb3842085f'.
```

**Comment by Etsija** [comment]
What do I need to build in order to test my UI branch against your PR?
- do I need to rebuild workers/config/docker image?
- do I need to run JIB, so ./gradlew JibDockerBuild?

File: docker-compose.yml [hunk 3]
```diff
@@ -186,6 +186,11 @@ services:
       ANALYZER_SENDER_TRANSPORT_QUEUE_NAME: analyzer_queue
       ANALYZER_SENDER_TRANSPORT_USERNAME: admin
       ANALYZER_SENDER_TRANSPORT_PASSWORD: admin
+      CONFIG_SENDER_TRANSPORT_TYPE: rabbitMQ
```
Comment (MarcelBochtler) [inline docker-compose.yml:189]: Commit message: `paramter`

**Review by MarcelBochtler — COMMENTED** [review]
LGTM. One small nit.

**Comment by Etsija** [comment]
From the UI project point of view, I can verify that this PR works: I am able to create an "empty" run (with empty job configuration) and verified that the Analyzer started, ran and finished successfully.

**Review by Etsija — APPROVED** [review]

**Comment by Etsija** [comment]
(BTW I can heartily recommend Git worktrees for this kind of tricky verifications where really two branches are needed: I just built the back-end straight out of checked out PR, then created a worktree where I ran the front-end on our UI dev branch.  Worked like a charm, no merge issues or anything.)

**Comment by mnonnenmacher** [comment]
> (BTW I can heartily recommend Git worktrees for this kind of tricky verifications where really two branches are needed: I just built the back-end straight out of checked out PR, then created a worktree where I ran the front-end on our UI dev branch. Worked like a charm, with good support for worktrees in VS Code.)

Interesting, never heard of that feature before.

**Review by MarcelBochtler — APPROVED** [review]

**Comment by sschuberth** [comment]
> Interesting, never heard of that feature before.

FYI, [worktrees](https://git-scm.com/docs/git-worktree) are a generic Git feature, no a VS Code specific feature. They basically are different checkouts for one local Git repo.

**Pull request merged**